File: svws-asd/src/main/java/de/svws_nrw/asd/validate/schule/ValidatorSchuleStammdaten.java
package de.svws_nrw.asd.validate.schule;

import de.svws_nrw.asd.data.CoreTypeException;
import de.svws_nrw.asd.types.schule.Schulform;
import de.svws_nrw.asd.validate.Validator;
import de.svws_nrw.asd.validate.ValidatorKontext;
import jakarta.validation.constraints.NotNull;

/**
 * Dieser Validator führt eine Statistikprüfung auf die Stammdaten einer Schule aus.
 */
public final class ValidatorSchuleStammdaten extends Validator {

	/**
	 * Erstellt einen neuen Validator mit den übergebenen Daten und dem übergebenen Kontext
	 *
	 * @param kontext   der Kontext des Validators
	 */
	public ValidatorSchuleStammdaten(final @NotNull ValidatorKontext kontext) {
		super(kontext);
	}

	@Override
	protected boolean pruefe() {
		boolean success = true;

		// Prüfe, ob die Schulform überhaupt gesetzt ist oder nicht
		final String schulformKrz = super.kontext().getSchuleStammdaten().schulform;
		success = exec(0, () -> (schulformKrz == null) || (schulformKrz.isBlank()), "Die Schulform muss gesetzt sein.");
		if (!success)
			return false;

		// Prüfe, ob die Schulform gültig gesetzt ist oder nicht
		success = exec(1, () -> {
			try {
				return Schulform.data().getWertByKuerzel(schulformKrz) == null;
			} catch (@SuppressWarnings("unused") final CoreTypeException e) {
				return false;
			}
		}, "Das Kürzel für die Schulform ist ungültig.");

		return success;
	}

}

File: svws-webclient/core/src/asd/validate/schule/ValidatorSchuleStammdaten.ts
import { Schulform } from '../../../asd/types/schule/Schulform';
import { Class } from '../../../java/lang/Class';
import { JavaString } from '../../../java/lang/JavaString';
import { ValidatorKontext } from '../../../asd/validate/ValidatorKontext';
import { Validator } from '../../../asd/validate/Validator';

export class ValidatorSchuleStammdaten extends Validator {


	/**
	 * Erstellt einen neuen Validator mit den übergebenen Daten und dem übergebenen Kontext
	 *
	 * @param kontext   der Kontext des Validators
	 */
	public constructor(kontext: ValidatorKontext) {
		super(kontext);
	}

	protected pruefe(): boolean {
		let success: boolean = true;
		const schulformKrz: string | null = super.kontext().getSchuleStammdaten().schulform;
		success = this.exec(0, { getAsBoolean: () => (schulformKrz === null) || (JavaString.isBlank(schulformKrz)) }, "Die Schulform muss gesetzt sein.");
		if (!success)
			return false;
		success = this.exec(1, { getAsBoolean: () => {
			try {
				return Schulform.data().getWertByKuerzel(schulformKrz) === null;
			} catch(e : any) {
				return false;
			}
		} }, "Das Kürzel für die Schulform ist ungültig.");
		return success;
	}

	transpilerCanonicalName(): string {
		return 'de.svws_nrw.asd.validate.schule.ValidatorSchuleStammdaten';
	}

	isTranspiledInstanceOf(name: string): boolean {
		return ['de.svws_nrw.asd.validate.schule.ValidatorSchuleStammdaten', 'de.svws_nrw.asd.validate.BasicValidator', 'de.svws_nrw.asd.validate.Validator'].includes(name);
	}

	public static class = new Class<ValidatorSchuleStammdaten>('de.svws_nrw.asd.validate.schule.ValidatorSchuleStammdaten');

}

export function cast_de_svws_nrw_asd_validate_schule_ValidatorSchuleStammdaten(obj: unknown): ValidatorSchuleStammdaten {
	return obj as ValidatorSchuleStammdaten;
}
